### PR TITLE
refactor: improve logging (add tracing crate)

### PR DESCRIPTION
**Pull request by axiomatic-aardvark**
Switch to using [tracing](https://crates.io/crates/tracing) instead of `println!()` for logging. 
Next steps:
- Add support for Prometheus (check out [this guide](https://tokio.rs/tokio/topics/tracing-next-steps))

### CODE REVIEW

File: src/gossip_agent/message_typing.rs [hunk 1]
```diff
@@ -159,7 +160,7 @@ impl GraphcastMessage {
             sig.to_string(),
         );
 
-        println!("{}{:#?}", "Encode message: ".cyan(), message,);
+        info!("{}{:#?}", "Encoded message: ".cyan(), message,);
```
Comment (hopeyen) [inline src/gossip_agent/message_typing.rs:163]: `Message` can be long, so maybe make it `debug` level?

**Review by hopeyen — DISMISSED** [review]
lgtm! is there a way to set log levels though command line? something like `cargo run --log_level=[info, debug, error,...]`?

**Comment by axiomatic-aardvark** [comment]
things left to do:
- enable debug log level
- pass in log level to WakuNodeConfig

File: src/gossip_agent/message_typing.rs (hunk 1 above)
Comment (axiomatic-aardvark) [reply src/gossip_agent/message_typing.rs]: ✅

**Review by hopeyen — APPROVED** [review]
🔥

**Review by hopeyen — APPROVED** [review]

**Review by hopeyen — APPROVED** [review]

**Pull request merged**